### PR TITLE
Finder unit tests

### PR DESCRIPTION
**Pull request by pfistererm**
Created extensive test cases for the finder classes (the central component doing the license scan).

## Type of request

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring 
- [ ] Documentation or documentation changes

## Concept 
More test coverage.
#### Description of the change
Added unit tests. Note that one test is ignored due to a bug in the LicenseScout (without ignoring it
the PR would not pass the CI build).

#### Initiator
Matthias Pfisterer

#### DB Changes (if so: incl. scripts with definitions and testdata)
		
None.

#### How has this Tested?
Running all tests with mvn clean install

## Checklist:

- [x] All the commits are signed.
- [x] My code follows the code style of this project.
- [x] I agree with die CLA.
- [x] I have read the CONTRIBUTING docs.
- [x] I have added/updated necessary documentation (if appropriate).

### CODE REVIEW

**Comment by davu-all** [comment]
Please fix the conflict and then i'll review. Thanks

**Comment by pfistererm** [comment]
@davu-all I resolved the conflict - please review again.

File: org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/BaseFinderTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    private ILFLog log;
+    private LicenseStoreData licenseStoreData;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        Logger.getGlobal().setLevel(Level.FINEST);
+        Handler consoleHandler = new ConsoleHandler();
+        consoleHandler.setLevel(Level.FINEST);
+        Logger.getGlobal().addHandler(consoleHandler);
+        // Handler[] handlers = Logger.getGlobal().getHandlers();
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/BaseFinderTest.java]: Could you please remove comment out code?

File: org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/CreateJarFileHelper.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+     * <p>This method can be used to collect the files to package with the
+     * method {@link #createJarArchiveZip(File, List, Path)}.</p> 
+     * 
+     * @param startDirectory
+     * @return a list of files
+     * @throws IOException
+     */
+    public static List<File> collectFiles(final File startDirectory) throws IOException {
+        final List<File> files = new ArrayList<>();
+        final SimpleFileVisitor<Path> visitor = new SimpleFileVisitor<Path>() {
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/CreateJarFileHelper.java]: Code cleaning: change to 
`final SimpleFileVisitor<Path> visitor = new SimpleFileVisitor<>() {`

File: org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/CreateJarFileHelper.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     * as the entry name in the ZIP file
+     * @throws IOException
+     */
+    public static void createJarArchiveZip(final File archiveFile, final List<File> tobePackagedFiles,
+                                           final Path basePath)
+            throws IOException {
+        try (final FileOutputStream fileOutputstream = new FileOutputStream(archiveFile);
+                final ZipOutputStream out = new ZipOutputStream(fileOutputstream)) {
+            for (final File tobePackaged : tobePackagedFiles) {
+                if (tobePackaged == null || !tobePackaged.exists() || tobePackaged.isDirectory())
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/CreateJarFileHelper.java]: Better readability: Please add the curly brackets to this if statement

File: org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/JavaJarFinderTest.java [hunk 4]
```diff
@@ -43,6 +62,355 @@ public void testPomResolutionUsed() {
...
+     * 
+     * <p>Tests with a packed JAR without license information and
+     * an inner packed JAR with a license file.</p>
+     * <p>Uses scan directory 'src/test/resources/scans/java-packed-with-inner-license-file'.</p>
+     * 
+     * @throws Exception 
+     */
+    @Test
+    public void testScanJavaPackedWithInnerLicenseFile() throws Exception {
+        {
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/JavaJarFinderTest.java]: [Optional] can we get rid of those brackets (here and in the following tests as well)?

File: org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/BaseFinderTest.java (hunk 1 above)
Comment (pfistererm) [reply org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/BaseFinderTest.java]: Changed.

File: org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/CreateJarFileHelper.java (hunk 2 above)
Comment (pfistererm) [reply org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/CreateJarFileHelper.java]: Changed.

File: org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/CreateJarFileHelper.java (hunk 3 above)
Comment (pfistererm) [reply org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/CreateJarFileHelper.java]: Changed.

File: org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/JavaJarFinderTest.java (hunk 4 above)
Comment (pfistererm) [reply org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/JavaJarFinderTest.java]: Changed.

**Pull request merged**